### ISSUE TITLE
Backward LSTM behavior mismatch

### ISSUE THREAD

**Issue by buddhapuneeth** [tensorflow/tensorflow#40345]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.2
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

v2.2.0-0-g2b96f3662b 2.2.0
**Describe the current behavior**
Say input is [a1 a2 a3] and running on GPU
In reverse LSTM, 
reversed_input_to_cudnn = [a3 a2 a1]
output_from_cudnn = [b3 b2 b1]
final output = [b1 b2 b3] 

This is based on the code and description mentioned here:
https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/keras/layers/recurrent_v2.py#L630
  
**Describe the expected behavior**
Expected final output is [b3 b2 b1] based on the description mentioned here:
https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/keras/layers/recurrent_v2.py#L1000

I'm not sure what internally is happening and what is the expected behavior?
Other inference engines like ONNXRuntime will give output [b1 b2 b3] and currently there is mismatch of outputs with ONNXRuntime and TF.

**Standalone code to reproduce the issue**
[simple_lstm.txt](https://github.com/tensorflow/tensorflow/files/4756947/simple_lstm.txt)
(rename to .h5) 
You can use this sample model to verify the results.

**Comment by jvishnuvardhan** [comment]
@buddhapuneeth Can you please share a simple standalone code to reproduce the issue? You could use google colab or Jupyter notebook. Thanks!

**Comment by buddhapuneeth** [comment]
code for reverse LSTM:
```
max_features = 10
maxlen = 10
model = Sequential()
model.add(Embedding(max_features, 128, input_length=maxlen))

model.add(LSTM(64, go_backwards=True, return_sequences = True))
model.add(Dense(1, activation='sigmoid'))
model.compile('adam', 'binary_crossentropy', metrics=['accuracy'])


from keras.models import load_model
model.save('simple_rev_lstm.h5')
```

To compare results with other inference engines like ONNXRuntime (by converting to ONNX)


```
# pip install onnxruntime
#env TF_KERAS=1
import numpy as np
import onnxruntime
from tensorflow.keras.models import load_model as load_model_tf_keras
np.random.seed(0)
input_data = np.random.randint(10, size=(2, 10))
# with CustomObjectScope({'GlorotUniform': glorot_uniform()}):
sess = onnxruntime.InferenceSession("simple_rev_lstm.onnx")
result = sess.run(["dense_1"], {'embedding_1_input': input_data.astype(np.float32)})
print("ONNX Runtime")
print(np.asarray(result[0]))

model = load_model_tf_keras('simple_rev_lstm.h5')
result = model.predict(input_data)
print("TF Runtime")
print(result)
```
OUTPUT:
```
ONNX Runtime
[[[0.5028585 ]
  [0.51010513]
  [0.5096171 ]
  [0.50567615]
  [0.50046337]
  [0.50202006]
  [0.5010086 ]
  [0.49436894]
  [0.4989641 ]
  [0.50032514]]

 [[0.50235206]
  [0.5045772 ]
  [0.504677  ]
  [0.5032916 ]
  [0.5016299 ]
  [0.50043035]
  [0.4991624 ]
  [0.50064933]
  [0.5028503 ]
  [0.5012636 ]]]

TF Runtime
[[[0.50032514]
  [0.4989641 ]
  [0.49436894]
  [0.5010086 ]
  [0.50202006]
  [0.50046337]
  [0.50567615]
  [0.5096171 ]
  [0.51010513]
  [0.5028585 ]]

 [[0.5012636 ]
  [0.5028503 ]
  [0.50064933]
  [0.4991624 ]
  [0.50043035]
  [0.5016299 ]
  [0.5032916 ]
  [0.504677  ]
  [0.5045773 ]
  [0.50235206]]]
```
Observation:
The values are in reverse to each other on axis 1.
Not sure which is correct behavior. Documentation is bit confusing here.

**Comment by buddhapuneeth** [comment]
@jvishnuvardhan any update? Let me know if you need more info'.

**Comment by hellomotor** [comment]
I'm confused too , clarification need.

**Comment by sushreebarsa** [comment]
@buddhapuneeth Could you please try on latest stable version of tf 2.5 or 2.4.1 and let us know if this is still an issue.Thanks!

**Comment by buddhapuneeth** [comment]
@sushreebarsa 
I am seeing same old issue on TF 2.5.0
Tried on Mac TF 2.5.0 (installed using pip)

**Comment by buddhapuneeth** [comment]
@sushreebarsa are you able to reproduce?
I am worried as bot marked this stall.

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
